### ISSUE TITLE
Move `internal/experimental/registry` out of `internal` package

### ISSUE THREAD

**Issue by owenthereal** [helm/helm#9188]
I would like to know whether it's possible to move https://github.com/helm/helm/tree/master/internal/experimental/registry out of the `internal` package so that project like https://github.com/hashicorp/terraform-provider-helm can implement OCI registry support by using Helm as a library.

Ref: https://github.com/hashicorp/terraform-provider-helm/issues/633

Output of `helm version`:

```
version.BuildInfo{Version:"v3.4.2", GitCommit:"23dd3af5e19a02d4f4baa5b2f242645a1a3af629", GitTreeState:"dirty", GoVersion:"go1.15.5"}
```

Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-19T08:38:20Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:47:43Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

N/A

**Comment by mattfarina** [comment]
This is the plan when it's ready. We are currently making some changes to the way OCI handling works in Helm. The next iteration will come out in 3.5.0. The code in the internal package is getting some changes that would be breaking if public to make this happen.

We have a proposal being worked on to move this out of the experiment phase at https://github.com/helm/community/blob/master/hips/hip-0006.md. Once this happens the package should be public to use.

**Comment by bryanisgreat** [comment]
hey, @mattfarina, do you have an idea of the roadmap for GA of this package yet (as current release is now v3.5.3)?

**Comment by bacongobbler** [comment]
https://github.com/helm/community/blob/master/hips/hip-0006.md#7-experimental-until-clear-messaging-from-oci

**Comment by mariadb-JeffBachtel** [comment]
@moolen weirdly, the distribution spec got a 1.0 release on May 5 - 2 days before the Helm Developer Call.

@mattfarina OCI (distribution spec) is 1.0. Are there further upstream issues to resolved before moving out of experimental in Helm?

**Comment by bacongobbler** [comment]
There is a LOT of work required before it can be unmarked experimental.

**Comment by bacongobbler** [comment]
I will also note that the OCI distribution spec is orthogonal to [the OCI Artifacts proposal](https://github.com/opencontainers/artifacts). While OCI distribution reaching 1.0 is nice, the latter is what is required before we can consider the experimental work "blessed" by the Open Container Initiative.

**Comment by leanrobot** [comment]
+1 looking forward to this feature. We are attempting to use gitlab OCI registry for storing our helm packages, and interfacing directly with helm package in go to install our charts into a cluster.

In the meantime, are there any recommended workarounds? Thoughts were either:
- Store images in standard helm repository
- Use `os.exec` to pull and update charts from OCI registries, install can happen through helm golang library.

**Comment by mariadb-JeffBachtel** [comment]
@bacongobbler OCI Artifacts repository will not be receiving a 1.0 (or any) version, please see https://github.com/opencontainers/artifacts/issues/23

**Comment by Ashtonian** [comment]
> This issue has been marked as stale because it has been open for 90 days with no activity. This thread will be automatically closed in 30 days if no further activity occurs.

badbot

**Comment by bacongobbler** [comment]
I'll add the "keep open" label so we can pin this issue as we have a few maintainers (and a few community members) working on this. Updates are being provided weekly on the developer call. If you'd like to get involved, [please join the developer call](https://github.com/helm/helm#community-discussion-contribution-and-support).

> can we [not](https://drewdevault.com/2021/10/26/stalebot.html)

Warning: rant ahead.

Helm is maintained by volunteers, often contributing to the project on personal or "work research" hours. We see 40+ issues opened per month.

We originally "left [the issue queue] to the community" to triage, but that made our job incredibly difficult. The core maintainers are responsible for planning the next release of Helm, which involves searching the issue queue for feature requests ready for review, follow up with any bug reports (and see if they need a fix), and try to answer questions as best as we can.

At one point we reached nearly 700 open issues. This was not sustainable - there are less than a handful of maintainers trying to plan the next Helm release, and we were spending the majority of our time searching through the issue queue just to find good candidates for the next release.

So we introduced the stale issue bot. Now we spend much less of our personal time searching/curating the issue queue and more time reviewing pull requests. Overall I'd call that a win.

On the question of "why close out feature requests?": if a feature request is left open for several years without a solution, closing it out due to inactivity is a fair response IMO. Many features are never implemented due to a lack of community interest, due to existing workarounds, or due to a lack of time and effort. If you want to see something implemented, [get involved](https://github.com/helm/helm/blob/main/CONTRIBUTING.md), hire a contractor, or find other like-minded individuals who are interested in collaborating. But this is not the place to dump a "project wishlist" and ask the maintainers to work on your ideas for free.

**Comment by Ashtonian** [comment]
Thank you for your contributions. 

> Warning: rant ahead.

I understand its usefulness, but automatically closing issues is a bandaid is doing as much harm as good. Also there are other things that can be done to improve ticket quality like making the submitter add additional data in a standardized format that can be easier to filter later.

**Comment by bacongobbler** [comment]
closing in favour of #10393.

**Comment by bacongobbler** [comment]
Err, rather it seems this is tracking the work to actually `mv internal/* pkg/`, whereas the latter is a meta issue to track the rest of the work necessary to do that. Re-opening.
